tensornetwork/ncon_interface.py

```diff
@@ -224,9 +428,50 @@ def _jittable_ncon(tensors: Sequence[Tensor], flat_labels: Tuple[int],
         con_order,
         np.intersect1d(con_order, contracted_labels, return_indices=True)[1])
 
-  # binary contractions
+  # contracted all positive labels appearing only once in `network_structure`
+  unique_labels, label_cnts = np.unique(
+      np.concatenate(network_structure), return_counts=True)
+  contractable_labels = unique_labels[np.logical_and(label_cnts == 1,
+                                                     unique_labels > 0)]
+  locs = np.sort(
+      np.nonzero([
+          np.any(np.isin(labels, contractable_labels))
+          for labels in network_structure
+      ])[0])
+
+  for loc in locs:
+    labels = network_structure[loc]
+    contractable_inds = np.nonzero(np.isin(labels, contractable_labels))[0]
+    network_structure[loc] = np.delete(labels, contractable_inds)
+    tensors[loc] = backend_obj.sum(tensors[loc], tuple(contractable_inds))
+
+  # update con_order after collapsing single labels
+  con_order = np.delete(con_order,
+                        np.nonzero(np.isin(con_order, contractable_labels))[0])
```

Comment:
I fear this isn't going to be nearly as efficient as using batch matrix multiplication on accelerated backends like JAX, especially since we're adding axes to the output tensor that just get summed over later.

Reply:
can you tell me which part you mean here?

Reply:
The backend functions I use to do batched matrix multiplications are e.g. `jax.numpy.matmul` for JAX, which supports broadcasting over a batch index.

the code above does simply perform a regular `sum` operation over positive labels that appear only once in the `ncon` call.

Reply:
Got it. this should be fine then.